### PR TITLE
[JIT] Don't throw when type is used in TorchScript

### PR DESCRIPTION
**Pull request by eellison**
Type objects in python have an attribute `__abstractmethods__` that throws when it is accessed, so we were failing with an AttributeError whenever a type was used in TorchScript. 

This pr prevents that error from happening. We can't just throw when a type is used because it could be used to access a static method: https://github.com/pytorch/pytorch/pull/27163

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -7320,6 +7320,14 @@ def test_call_python(a):
 
             self.assertEqual(cu.test_call_python(*inputs), outputs)
 
+    def test_type_call_in_script(self):
+        @torch.jit.script
+        def fn(x):
+            return type(x)
+
+        with self.assertRaisesRegex(RuntimeError, "value of type type"):
```
Comment (driazati) [inline test/test_jit.py]: Can you put quotes around the output here so it doesn't look like a typo?
e.g.
`value of type 'type'`

File: torch/_jit_internal.py [hunk 2]
```diff
@@ -140,7 +140,13 @@ def can_compile_class(cls):
...
     if is_ignored_fn(cls):
         return False
-    fns = [getattr(cls, name) for name in cls.__dict__ if inspect.isroutine(getattr(cls, name))]
+    names = cls.__dict__
+    # '__abstractmethods__' errors when accessed as an AttributeError, so
+    # exit before calling getattr below. This can occur in a type object or class
+    # with an abstract method, in either case we cannot compile it as a class
+    if '__abstractmethods__' in names:
+        return False
+    fns = [getattr(cls, name) for name in names if inspect.isroutine(getattr(cls, name))]
```
Comment (driazati) [inline torch/_jit_internal.py]: You can avoid special casing `type` here by changing the `getattr` to `getattr(cls, name, None)`

Comment (driazati) [reply torch/_jit_internal.py]: Or better yet change this whole thing to this which doesn't have the problem of accessing the attribute at all:

```python
members = inspect.getmembers(cls)
members = map(lambda x: x[1], members)
fns = filter(inspect.isroutine, members)
```

**Comment by suo** [comment]
Just checking in here—any blockers?

**Review by driazati — APPROVED** [review]

**Pull request closed without merging**